### DISCUSSION TITLE
Email Embed Support?

### DISCUSSION THREAD

**Discussion by blopker** [add2cal/add-to-calendar-button#417] [Ideas / Requests]
First off, this is a great lib, thanks for putting this together!

I wanted to open this discussion around how to embed the Add To Calendar button in an email. I know it relies on JavaScript to do its thing, but maybe there's a way to have it render pure email-friendly HTML? 

I saw the issue here: https://github.com/add2cal/add-to-calendar-button/issues/219. It sounds like there might be some hope. How's that going?

Unfortunately, we use Python on the backend, and not Node. However, even seeing what this library might output as HTML only would help us greatly.

Thank you!

**Comment by alsage13** [comment]
I 100% would love this. Any idea when this might roll out? I would happily donate a bit to the cause to help make this real. :)

**Reply by jekuer** [reply]
4-6 weeks as per the current planning 🙂

**Comment by alsage13** [comment]
Excellent, thanks so much!

**Comment by vanquishkuso** [comment]
Has this been done yet? I tried to find information about this but couldn't find any.

**Reply by jekuer** [reply]
Yes and no.

With https://add-to-calendar-pro.com/, you get an option to copy direct linked buttons and/or images into your emails.
However, there is no option the Add to Calendar Button (this lib here) would work in emails. This is technically not possible. With the upcoming v3 (next week), it will be possible to render some initial HTML server-side (for SSR prerendering). Very theoretically, this could be used for this use case, but please don't - you would be faster to rebuild the HTML yourself by hand.
We are planning for an additional service to provide free email links (that actually work). Expect this over the next 4 months.

Generally speaking: Mind that you can only put static information into emails!
Therefore, you would need to put a static link there. Like those links you can get from all kinds of "calendar link generators". Problem: They do not work for many people, because the same link behaves differently on different devices, operating systems, and so on. 
You need a static link to a middleware, that checks the user and then directs them to the respective calendar endpoint. That's what we are doing with Add to Calendar PRO. That's what is technically impossible with the Add to Calendar Button (as it is no service, but a component rendered on the client side).
